qtconsole/mainwindow.py

```diff
@@ -809,7 +810,11 @@ def set_syntax_style(self, syntax_style):
         self.active_frontend._syntax_style_changed()
         self.active_frontend._style_sheet_changed()
         self.active_frontend.reset(clear=True)
-        self.active_frontend._execute("%colors linux", True)
+        self.active_frontend._execute(
+            f"from IPython.core.ultratb import VerboseTB;"
+            "VerboseTB._tb_highlight_style = '{syntax_style}'",
```

Comment:
A couple of comments about this change:

* The `_tb_highlight_style` was added in IPython 8.15 (see https://github.com/ipython/ipython/pull/14138). So, it's necessary a check that executes the new code for that or more recent versions, and leaves the previous one for older versions.
* It'd be a good idea to make `_tb_highlight_style` a public attribute of `VerboseTB` by submitting a PR to IPython. That would require an additional check here but it'd prevent they breaking us in the future.

Reply:
I think something like this could work:

```suggestion
f"""
from IPython.core.ultratb import VerboseTB
try:
    VerboseTB._tb_highlight_style = '{syntax_style}'
except AttributeError:
    get_ipython().run_line_magic('colors', '{colors}')
""",
```

Reply:
This is very good approach, thanks @dalthviz! However, given that `_tb_highlight_style` is a private attribute of `VerboseTB`, I think it'd be a good idea to promote it to a public one in IPython and let Matthias know that we're using it here.

Otherwise, this functionality could break easily without us being aware of it (given that now there's a fallback in case that attribute is no longer available).

Reply:
Created https://github.com/ipython/ipython/issues/14464 to check your suggestion @ccordoba12

Reply:
Yep, thanks! I saw Matthias answer, so now please proceed to implement his suggestions there.